software/control/filterwheel.py

```diff
@@ -0,0 +1,64 @@
+import os
+import time
+
+from control._def import *
+
+class SquidFilterWheelWrapper:
+
+    def __init__(self, navigationController = None):
+        # emission filter position
+        self.w_pos_index = 0
+
+        self.navigationController = navigationController
+
+    def homing(self):
+        # make sure the start homing position is not at homing position  
+        starttime = time.time() 
+        self.navigationController.home_w()
+        self.navigationController.wait_till_operation_is_completed()
+        # judge whether the current position is a homing position or not
+        if time.time() - starttime > 0.02:
+            # move offset, could be modified later
+            self.navigationController.move_w(SQUID_FILTERWHEEL_OFFSET)
+
+        self.w_pos_index = 0
+        
+    def next_position(self):
+        if self.w_pos_index < SQUID_FILTERWHEEL_MAX_INDEX:
+            self.navigationController.move_w(SCREW_PITCH_W_MM / 8.0)
```

Comment:
Where does the `/ 8.0' factor come from here?  If it's the same as the rest of the bare 8.0, can you pull it out into a class constant with a clear name and maybe a comment?  EG:
```
class SquidFilterWheel:
    # Comment explaining why this exists and why it might be 8
    SCREW_PITCH_FRACTION_PER_INDEX = 8.0
```

Also it looks like it might just be converting to a fraction of the full screw pitch via knowing that SQUID_FILTERWHEEL_MAX_INDEX == 7.  Should it instead be tied to that constant, instead of hard coded to 8?

Reply:
(same for all occurrences of 8 below)

Reply:
@veerwang

Reply:
OK